app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -443,16 +375,122 @@ private boolean appNotificationScheduleActive(String appPackageName) {
 
         int startTime = appSelection.getStartTime();
         int stopTime = appSelection.getStopTime();
-        Date currDate = new Date();
 
         // Get current time
         Calendar cal = Calendar.getInstance();
-        cal.setTime(currDate);
         int hour = cal.get(Calendar.HOUR_OF_DAY);
         int minute = cal.get(Calendar.MINUTE);
-        int currTime = hour*60 + minute;
+        int currTime = hour * 60 + minute;
+
+        return ((currTime >= startTime) && (currTime <= stopTime));
+    }
+
+    private boolean isScreenOn() {
+        PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
 
-        return ((currTime >= startTime) && (currTime < stopTime));
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT) {
+            // API >= 20
+            return pm.isInteractive();
+        }
+
+        // API <= 19, use deprecated
+        //noinspection deprecation
+        return pm.isScreenOn();
+    }
+
+    /**
+     * Checks if any of the <code>CharSequence</code> items contains the provided <code>filter</code>
+     * text.
+     */
+    private static boolean anyMatchesFilter(String filter, CharSequence ... items) {
+        if (filter != null && !filter.isEmpty()) {
+            String[] parts = filter.split("\\s*,\\s*");
+
+            for (CharSequence item : items) {
+                if (item != null) {
+                    String tmp = item.toString();
+                    for (String filterText : parts) {
+                        if (filterText.length() > 0 && tmp.contains(filterText)) {
```

Comment:
Trim filterText. Otherwise, someone may type this as a filter `,     ,    ,` that is essentially empty text and you will be looking for whitespaces. Note sure if this regular expression `\\s*,\\s*` handles whitespace, if so then that is fine.